src/TripSegment/index.js

```diff
@@ -0,0 +1,282 @@
+// @flow
+import * as React from "react";
+import styled, { keyframes, css } from "styled-components";
+
+import ChevronUp from "../icons/ChevronUp";
+import ChevronDown from "../icons/ChevronDown";
+import defaultTokens from "../defaultTokens";
+import CarrierLogo, { StyledCarrierLogo } from "../CarrierLogo";
+import Airplane from "../icons/Airplane";
+import Bus from "../icons/Bus";
+import Train from "../icons/Train";
+import Text from "../Text";
+import { Item } from "../List/ListItem";
+
+import type { Props, State } from "./index";
+
+// TODO: Fix animations
+const ToggleUp = keyframes`
+  from { max-height: 0%; overflow: hidden; opacity: 0 }
+  to { max-height: 100%; overflow: visible; top: 0; opacity: 1 }
+`;
+
+const ToggleDown = keyframes`
+  0% { max-height: 0;  overflow: hidden; opacity: 1}
+  75% { opacity: 0.4; max-height: 50% }
+  100%  { max-height: 100%; overflow: visible; opacity: 0}
+`;
+
+const Milestone = styled.div`
+  display: flex;
+  height: 55px;
+  align-items: center;
+  padding-right: ${({ theme }) => theme.orbit.spaceXSmall};
+`;
+
+Milestone.defaultProps = {
+  theme: defaultTokens,
+};
+
+export const StyledSegment = styled(({ className, children }) => (
```

Comment:
`Text` component used for cities and times should have line-height: 1.2 (hard-code it), otherwise the TripSegment is 6px larger.